src/app/_components/Feedback/Feedback.js

```diff
@@ -0,0 +1,146 @@
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import LoveIcon from '@icons/loveIcon.svg';
+import { useCreateFeedback } from '@/app/_hooks';
+import SendFeedback from '@/lib/validationSchemas/sendFeedbackSchema';
+import { Modal } from '../Modal';
+import { TextInputField } from '../InputFields';
+import { CheckBox } from '../CheckBox';
+import { TextArea } from '../TextArea';
+import { PillButton } from '../PillButton';
+import { cn } from '@/utils/cn';
+import { buttonColorVariant, buttonVariant } from '../PillButton/style';
+
+const MAX_LENGTH = 320;
+
+export function Feedback({ isFeedbackOpen, onClose }) {
+  const [isFormOpen, setFormOpen] = useState(true);
+
+  const [name, setName] = useState('');
+  const [phone, setPhone] = useState('');
+  const [email, setEmail] = useState('');
+  const [isChecked, setChecked] = useState(true);
+  const [message, setMessage] = useState('');
+
+  const [validationErrors, setValidationErrors] = useState({});
+
+  const { mutate: createFeedback } = useCreateFeedback();
+
+  const onSubmit = e => {
+    e.preventDefault();
+    const formData = {
+      name,
+      phone,
+      callMe: isChecked,
+      email: email.length ? email : undefined,
+      message,
+    };
+
+    try {
+      SendFeedback.parse(formData);
+      setValidationErrors({});
+      createFeedback(formData);
+
+      setFormOpen(false);
+    } catch (error) {
+      const errors = {};
+      error.errors.forEach(validationError => {
```

Comment:
🟠 instead of mutating object in `forEach` consider using `map` and `reduce` to follow immutable approach

Reply:
**Is the mutable approach bad in this case?**

_1 forEach - It simply iterates over the elements. **( My case )**
2  map: It creates a new array by applying a function to each element of the original array. 
3 reduce:  when you need to accumulate a result._

Reply:
it is a general rule/practise that any kind of mutation is 'bad' - as it is complicated, and leaves less traces to debug, comparing to immutable approach

you could also say that from philosophical point of view that 'forEach' is _consuming_ data, rather than used as an any iterator to do code you want inside on each item

map/reduce on the other hand are data modificators - they are suppose to change shape of data, so these are preferred to use when you want to transform data from one shape to another

Reply:
Аргументована відповідь. 
Добавив відповідні зміни.